src/Rakaposhi.Core/Rakaposhi.UnitTest/RoleTest.cs

```diff
@@ -0,0 +1,245 @@
+        private static IEnumerable<object[]> roles
+        {
+            get
+            {
+                return new[]
+                {
+                    new object[]
+                    {
+                        new List<Role>()
+                        {
+                            new Role()
+                            {
+                                RecId = 1,
+                                Name = "User1",
+                                Description ="User1Description"
+                            },
+                            new Role()
+                            {
+                                RecId = 2,
+                                Name = "User2",
+                                Description = "User2Description"
+                            },
+                            new Role()
+                            {
+                                RecId = 3,
+                                Name = "User3",
+                                Description = "User3Description"
+                            }
+                        }
+                    }
+                };
+            }
+        }
+
+        public RoleTest()
+        {
+            IRepositoryFactory factory = new FakeDBRepositoryFactory();
+            RoleService roleService = new RoleService(factory);
+            _controller = new RoleController(roleService);
+        }
+
+
+        [DataTestMethod]
+        [DataRow(1, "User1", "User1Description")]
+        [DataRow(1, "User2", "User2Description")]
+        [DataRow(1, "User3", "User3Description")]
+
+        [TestMethod]
+        public void RoleCreate_Valid(long id, string name, string description)
+        {
+            //Arrange
+            var role = new Role()
+            {
+                RecId = id,
+                Name = name,
+                Description = description
+            };
+
+            //Act
+            var response = _controller.Create(role) as CreatedResult;
+            var newRole = (Role)response.Value;
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.Created, (HttpStatusCode)response.StatusCode);
+            Assert.IsNotNull(value: newRole);
+            Assert.AreEqual(expected: role.RecId, actual: newRole.RecId);
+            Assert.AreEqual(expected: role.Name, actual: newRole.Name);
+            Assert.AreEqual(expected: role.Description, actual: newRole.Description);
+        }
+
+        [TestMethod]
+        public void RoleCreate_InValid()
+        {
+            //Arrange
+            var role = new Role()
+            {
+                RecId = null,
+                Name = "User1"
+            };
+
+            //Act
+            var response = _controller.Create(role) as BadRequestObjectResult;
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.BadRequest, (HttpStatusCode)response.StatusCode);
+            Assert.IsTrue(condition: ErrorCode.ADDERROR == response.Value.ToString());
```

Comment:
Can you please change IsTrue to AreEqual?

Reply:
fixed.